### PR TITLE
Add setup for running tests in drone CI

### LINKED ISSUE #644: Drone setup
Move from Travis to Drone

### PR DESCRIPTION
**Pull request by dpakach**
## drone setup
Fixes  #644

- Add make targets for running tests
- Add drone.yml file

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,383 @@
+workspace:
+  base: /var/www/owncloud
+  path: apps/contacts
+
+branches: [master, release*, release/*]
+
+pipeline:
+  install-server:
```
Comment (phil-davis) [inline .drone.yml]: now we call this ``install-core``
look in passowrd_policy or files_texteditor for an example of all the "standard" stuff.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+    version: ${OC_VERSION}
+    pull: true
+    db_type: ${DB_TYPE}
+    db_name: ${DB_NAME}
+    db_host: ${DB_TYPE}
+    db_username: autotest
+    db_password: owncloud
+    when:
+      matrix:
+        NEED_SERVER: true
```
Comment (phil-davis) [inline .drone.yml]: We put ``NEED_CORE`` here

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+      - make test-php-style
+    when:
+      matrix:
+        TEST_SUITE: owncloud-coding-standard
+
+  phpunit-tests:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    environment:
+      - PHP_VERSION=${PHP_VERSION}
```
Comment (phil-davis) [inline .drone.yml]: this line is not needed

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+      - if [ "${COVERAGE}" = "true" ]; then make test-php-unit-dbg; fi
+    when:
+      matrix:
+        TEST_SUITE: phpunit
+
+  integration-tests:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    environment:
+      - PHP_VERSION=${PHP_VERSION}
```
Comment (phil-davis) [inline .drone.yml]: this line is not needed

File: .drone.yml [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+      - make test-js
+    when:
+      matrix:
+        TEST_SUITE: js
+
+  codecov:
+    image: plugins/codecov:2
+    secrets: [codecov_token]
+    pull: true
+    files:
```
Comment (phil-davis) [inline .drone.yml]: Instead of ``files`` section, we have been putting:
```
    paths:
      - tests/output/*.xml
```

File: .drone.yml [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+      TEST_SUITE: js
+      COVERAGE_JS: true
+
+    # Unit Tests
+    - PHP_VERSION: 7.1
+      OC_VERSION: daily-master-qa
+      TEST_SUITE: phpunit
+      DB_TYPE: sqlite
+      DB_NAME: owncloud
+      NEED_SERVER: true
```
Comment (phil-davis) [inline .drone.yml]: this now becomes ``NEED_CORE`` in each ``phpunit`` matrix entry

File: .gitignore [hunk 7]
```diff
@@ -5,3 +5,22 @@ js/vendor
 coverage
 npm-debug.log
 package-lock.json
+
+.php_cs.cache
+
+# Composer
+composer.phar
```
Comment (phil-davis) [inline .gitignore]: We should not have ``composer.phar`` "accidentally" loaded from anywhere.
So this should not be needed.

File: .gitignore [hunk 8]
```diff
@@ -5,3 +5,22 @@ js/vendor
...
+
+# Mac OS
+.DS_Store
+
+# Tests - auto-generated files
+/tests/acceptance/output*
+/tests/output/
+
+# Node.js modules
+node_modules/
```
Comment (phil-davis) [inline .gitignore:23]: Line 1 can be deleted - this duplicates it

File: Makefile [hunk 9]
```diff
@@ -62,6 +62,10 @@ endif
 endif
 endif
 
+# bin file definitions
+PHPUNIT=php -d zend.enable_gc=0 "$(CURDIR)/../../lib/composer/bin/phpunit"
```
Comment (phil-davis) [inline Makefile]: In other repos we put ``PWD`` (and in the next line)
Why is this a bit different?

File: Makefile [hunk 10]
```diff
@@ -205,3 +196,72 @@ endif
 ifneq (,$(wildcard $(CURDIR)/package.json))
 	$(npm) run watch
 endif
+
+##
+## Tests
+##--------------------------------------
+
+.PHONY: test-syntax-php
+test-syntax-php:
```
Comment (phil-davis) [inline Makefile]: I don't think this is needed. Code style checks effectively do lint syntax checks for us.

File: phpunit.xml [hunk 11]
```diff
@@ -1,7 +1,29 @@
-<phpunit bootstrap="tests/bootstrap.php" colors="true">
```
Comment (phil-davis) [inline phpunit.xml:1]: can we delete ``tests/bootstrap.php`` - I think it is not needed any more?

**Review by phil-davis — COMMENTED** [review]
Various comments about standardizing this even more like other apps.\
This will be great to get it sorted out!

File: Makefile [hunk 12]
```diff
@@ -62,6 +62,10 @@ endif
 endif
 endif
 
+# bin file definitions
+PHPUNIT=php -d zend.enable_gc=0 "$(PWD)/../../lib/composer/bin/phpunit"
+PHPUNITDBG=phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "$(PWD)/../../lib/composer/bin/phpunit"
+PHP_CS_FIXER=php -d zend.enable_gc=0 vendor-bin/owncloud-codestyle/vendor/bin/php-cs-fixer
```
Comment (skshetry) [inline Makefile:68]: phpstan/phan

File: Makefile [hunk 13]
```diff
@@ -205,3 +196,66 @@ endif
...
+	$(PHP_CS_FIXER) fix -v --diff --diff-format udiff --allow-risky yes
+
+.PHONY: test-codecheck
+test-codecheck: test-syntax-php
+	$(occ) app:check-code $(app_name) -c private -c strong-comparison
+
+.PHONY: test-codecheck-deprecations
+test-codecheck-deprecations:
+	$(occ) app:check-code $(app_name) -c deprecation
+
```
Comment (skshetry) [inline Makefile:246]: add `test-php-phpstan` and `test-php-phan`

File: Makefile [hunk 14]
```diff
@@ -205,3 +196,66 @@ endif
...
+#
+# Dependency management
+#--------------------------------------
+composer.lock: composer.json
+	@echo composer.lock is not up to date.
+
+vendor: composer.lock
+	composer install --no-dev
+
+vendor/bin/phpunit: composer.lock
```
Comment (skshetry) [inline Makefile]: this is not needed

**Review by skshetry — DISMISSED** [review]

File: .drone.yml [hunk 15]
```diff
@@ -0,0 +1,381 @@
...
+  fix-permissions:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - chown www-data /var/www/owncloud -R
+      - chmod 777 /var/www/owncloud/tests/acceptance/filesForUpload -R
+      - chmod +x /var/www/owncloud/tests/acceptance/run.sh
+    when:
+      matrix:
+        NEED_CORE: true
```
Comment (phil-davis) [inline .drone.yml]: This is usually ``NEED_SERVER``

File: .drone.yml [hunk 16]
```diff
@@ -0,0 +1,381 @@
...
+
+  owncloud-log:
+    image: owncloud/ubuntu:16.04
+    detach: true
+    pull: true
+    commands:
+      - tail -f /var/www/owncloud/data/owncloud.log
+    when:
+      matrix:
+        NEED_CORE: true
```
Comment (phil-davis) [inline .drone.yml]: This is usually ``NEED_SERVER``

File: .gitignore [hunk 17]
```diff
@@ -1,7 +1,24 @@
-node_modules
 build
 js/public
 js/vendor
 coverage
```
Comment (phil-davis) [inline .gitignore:4]: I think we can delete this line

Comment (dpakach) [reply .gitignore:4]: That line is for js coverage files so I dont think we can delete that.
Or do we move that to somewhere else What do you think??

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Looks good. Actually there is only 1 PHP unit test case, and 1 integration test case! Coverage can't be too good.

I will merge this. Then we can remove travis in a next PR.

**Pull request merged**